hs_core/models.py

```diff
@@ -1925,7 +1925,8 @@ def delete(self, using=None):
                 # elements
                 fl.logical_file.metadata.delete()
             # COUCH: delete of file objects now cascades.
-            fl.delete()
+            if fl.reference_file_path is None:
+                fl.delete()
```

Comment:
@ismagilani I think for reference file, the reference ResourceFile needs to be deleted also. Otherwise, there would be left over ResourceFile object detached from the resource that has been deleted.

Reply:
I am a little confused since this particular piece of code is dealing with deletion of physical files not ResourceFile objects.  That's why I added the conditional to not delete the physical reference files.  I do not see anywhere in the code where we are deleting ResourceFile objects specifically.  Isn't that taken dare of the delete call further down in this method which calls the delete on the super class.  I think that's a cascading delete and garbage collects the ResourceFile objects as well when the parent Resource is deleted.

Reply:
*taken care of

Reply:
Ok, you are right. Removing that conditional still keeps the reference file in place so it's not needed.